core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -32,6 +50,7 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
 
     sealed class ResolveResult {
```

Comment:
That somewhat maps to current API in a following way:
```
data class ResolutionResult(
    val dependencies: ScriptDependencies?,
    val diagnostics: List<ScriptDiagnostic>
) {
    val isSuccess get() = dependencies != null
}
```

Not sure why we should avoid hierarchies. Can you explain your reasoning a bit?